truss/cli/cli.py

```diff
@@ -414,6 +414,14 @@ def push(
     # TODO(Abu): This needs to be refactored to be more generic
     service = remote_provider.push(tr, model_name, publish=publish, trusted=trusted)  # type: ignore
 
+    # Log a warning if using secrets without --trusted.
+    if tr.spec.config.secrets and not trusted:
+        not_trusted_text = """
+Warning: your Truss has secrets but was not pushed with --trusted.
+Please push with --trusted to grant access to secrets.
+"""
+        click.echo(not_trusted_text)
```

Comment:
Feedback on messaging, formatting, etc. welcome!

I could also stack this PR on top of https://github.com/basetenlabs/truss/pull/717 and use `console.print(not_trusted_text, style=yellow)`.

Reply:
I just merged https://github.com/basetenlabs/truss/pull/717, so you can merge master and use the colors

One other thought here, how does this look if we put it after the "Model was successfully pushed" copy? I worry that it gets lost if it's above the big block of development model text

Reply:
Done, I think both changes help!